core/src/main/java/org/elasticsearch/gateway/DanglingIndicesState.java

```diff
@@ -104,41 +116,85 @@ void cleanupAllocatedDangledIndices(MetaData metaData) {
     }
 
     /**
-     * Finds (@{link #findNewAndAddDanglingIndices}) and adds the new dangling indices
-     * to the currently tracked dangling indices.
+     * Finds (@{link #findDanglingAndDeletedIndices}) and adds the new dangling indices
+     * to the currently tracked dangling indices, while returning the indices to be deleted.
      */
-    void findNewAndAddDanglingIndices(MetaData metaData) {
-        danglingIndices.putAll(findNewDanglingIndices(metaData));
+    Map<Index, IndexMetaData> findOnDiskAndAddDanglingIndices(MetaData metaData) {
+        Tuple<Map<Index, IndexMetaData>, Map<Index, IndexMetaData>> result = findDanglingAndDeletedIndices(metaData);
```

Comment:
I think we can avoid returning tuples of maps here? I took a [stab](https://gist.github.com/areek/f7ebb27ab0b79d0bb422) at separating out logic for loading dangled index states and the logic for treating them as dangled/delete indices. WDYT?